astropy/coordinates/tests/test_angles.py

```diff
@@ -1092,3 +1092,33 @@ def test_str_repr_angles_nan(cls, input, expstr, exprepr):
     # Deleting whitespaces since repr appears to be adding them for some values
     # making the test fail.
     assert repr(q).replace(" ", "") == f'<{cls.__name__}{exprepr}>'.replace(" ","")
+
+
+@pytest.mark.parametrize(
+    "value,expected_value,dtype",
+    [
+        (np.pi / 2, np.pi / 2, np.float64),
+        (np.float32(np.pi / 2), np.float32(np.pi / 2), np.float32),
+        (np.float32(np.pi / 2), np.pi / 2, np.float64),
```

Comment:
This and the case below are still failing, since I didn't yet implement the coercion of these values.